prerequisites/build-functions/build_and_install.sh

```diff
@@ -5,6 +5,8 @@
 # shellcheck source=prerequisites/build-functions/edit_GCC_download_prereqs_file_if_necessary.sh
 source "${OPENCOARRAYS_SRC_DIR}/prerequisites/build-functions/edit_GCC_download_prereqs_file_if_necessary.sh"
 
+function version { echo "$@" | awk -F. '{ printf("%d%03d%03d%03d\n", $1,$2,$3,$4); }'; }
```

Comment:
It would be nice to have bash-only solutions, but this is so easy to do with awk that it's tough to make a case for replacing this.

Reply:
Right.  I considered that, but the bash-only solution I found is [31 lines long] versus this one-liner and `awk` is part of the [Linux Standard Base] and [XCode] so it seems safe to assume it will always be present on the systems of interest.   Of course, performance is another matter, but as always, I'd have to see some data to choose the longer solution over the short, clear one.

[XCode]: https://developer.apple.com/xcode/features/
[31 lines long]: https://stackoverflow.com/questions/4023830/how-to-compare-two-strings-in-dot-separated-version-format-in-bash
[Linux Standard Base]: https://en.wikipedia.org/wiki/AWK#History